smt/applications/mixed_integer.py

```diff
@@ -0,0 +1,256 @@
+"""
+Author: Remi Lafage <remi.lafage@onera.fr>
+
+This package is distributed under New BSD license.
+"""
+import numpy as np
+from smt.surrogate_models.surrogate_model import SurrogateModel
+from smt.sampling_methods.sampling_method import SamplingMethod
+
+FLOAT = "cont"
+INT = "int"
+ENUM = "cate"
+
+
+def check_xspec_consistency(xtypes, xlimits):
+    if len(xlimits) != len(xtypes):
+        raise ValueError(
+            "number of x limits ({}) do not"
+            "correspond to number of specified types ({})".format(
+                len(xlimits), len(xtypes)
+            )
+        )
+
+    for i, xtyp in enumerate(xtypes):
+        if (not isinstance(xtyp, tuple)) and len(xlimits[i]) != 2:
+            raise ValueError(
+                "Bad x limits ({}) for variable type {} (index={})".format(
+                    xlimits[i], xtyp, i
+                )
+            )
+
+        if (
+            xtyp != FLOAT
+            and xtyp != INT
+            and (not isinstance(xtyp, tuple) or xtyp[0] != ENUM)
+        ):
+            raise ValueError("Bad type specification {}".format(xtyp))
+
+        if isinstance(xtyp, tuple) and len(xlimits[i]) != xtyp[1]:
+            raise ValueError(
+                "Bad x limits and x types specs not consistent. "
+                "Got a categorical type with {} levels "
+                "while x limits contains {} values (index={})".format(
+                    xtyp[1], len(xlimits[i]), i
+                )
+            )
+
+
+def compute_x_unfold_dimension(xtypes):
+    res = 0
+    for xtyp in xtypes:
+        if xtyp == FLOAT or xtyp == INT:
+            res += 1
+        elif isinstance(xtyp, tuple) and xtyp[0] == ENUM:
+            res += xtyp[1]
+        else:
+            raise ValueError(
+                "Bad var type specification: "
+                "should be 'cont', 'int' or ('cate', n), got {}".format(xtyp)
```

Comment:
I don't understand if the user should write as before (int, cont, cate) or INT, FLOAT, ENUM

Reply:
I've defined INT="int", FLOAT="cont" and ENUM="cate" variables in mixed_integer (I think float and enum are more common than cont and cate). I've to change the error messages. The idea is to use constant variables instead of strings because if you make a typo (ex. FLOT) you get an error message (otherwise the string "flot" will pass).

Reply:
Ok, then, it works with either "cont" or FLOAT. Yes, I understand your point of view, these are more common name in computer science/programming.